### PR TITLE
Extend user_context with additional data

### PR DESCRIPTION
**Pull request by andehen**
This context is added on most events (everything except page pings and perhaps page/screen views). Any other data you can think of we should add?

### CODE REVIEW

**Comment by poudrouxj** [comment]
With price data this begs perhaps for some localization (e.g. currencies) any where we could add this on the event as well perhaps?

File: user_context.json [hunk 1]
```diff
@@ -27,6 +27,36 @@
...
+      "description": "The gross price of the cart.",
+      "type": "number"
+    },
+    "num_recipes_in_cart": {
+      "description": "Current number of recipes in the cart.",
+      "type": "integer",
+      "maximum": 9999999999,
+      "minimum": 0
+    },
+    "delivery_slot_selected": {
```
Comment (fredrijo) [inline user_context.json:57]: Would it make sense to add _which_ slot was selected too?

Comment (andehen) [reply user_context.json:57]: Yeah! Instead of a bool, we can add `delivery_slot_id`

**Comment by vegarped** [comment]
Nice! Would it make sense to add `habit_segment` here as well? Or is that better to merge in with dbt based on user_id and timestamp later?

**Comment by andehen** [comment]
> Nice! Would it make sense to add `habit_segment` here as well? Or is that better to merge in with dbt based on user_id and timestamp later?

I think that would be better to enrich later. We likely just want to include data that's only available in the moment the event happens (or that's hard to infer later). We are going to need an enrichment process in Snowflake anyway and there it makes sense to include stuff like habit segment.

**Review by vegarped — APPROVED** [review]

File: user_context.json [hunk 2]
```diff
@@ -23,10 +23,55 @@
...
+      "type": "array",
+      "items": {
+        "description": "Product id",
+        "type": "string",
+        "maxLength": 128
+      },
+      "maxItems": 9999999999
+    },
+    "net_price_cart": {
+      "description": "The gross price of the cart.",
```
Comment (marcus-gt) [inline user_context.json:53]: Should also be updated to net :)

**Review by marcus-gt — APPROVED** [review]
Great stuff, Anders!

**Comment by poudrouxj** [comment]
@andehen : Does it make sense to add the consent flags in this context? I was thinking for historical bookkeeping it may be a nice feature to have in snowflake as well 🤔

**Comment by andehen** [comment]
> @andehen : Does it make sense to add the consent flags in this context? I was thinking for historical bookkeeping it may be a nice feature to have in snowflake as well thinking

Yeah, that could make sense. Especially the cookie consent setting. However, I won't have time to implement that in Tienda now (need to get this out), so will have to wait for the next iteration on this.

**Pull request merged**